WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift

```diff
@@ -373,9 +373,9 @@ private extension StoreStatsV4PeriodViewController {
         let storageManager = ServiceLocator.storageManager
         let dateFormatter = DateFormatter.Stats.statsDayFormatter
         dateFormatter.timeZone = siteTimezone
-        let predicate = NSPredicate(format: "granularity ==[c] %@ AND date == %@",
+        let predicate = NSPredicate(format: "granularity ==[c] %@ AND timeRange == %@",
                                     timeRange.siteVisitStatsGranularity.rawValue,
-                                    dateFormatter.string(from: currentDate))
+                                    timeRange.rawValue)
```

Comment:
Could the new Predicates be used here since we are changing this NSPredicate? p91TBi-3BJ

Reply:
I guess we could, but we would need to `import storage`, how do we feel about that? cc @woocommerce/ios-developers

Reply:
I actually tried that earlier, but importing `Storage` resulted in a bunch of ambiguous types in `StoreStatsV4PeriodViewController` 😞 If not importing `Storage`, I'm not sure how to use the operators like `public func && (p1: TypedPredicate, p2: TypedPredicate) -> CompoundPredicate` 😐 

It'd be nice if we have a separate foundation framework with helpers that are applicable to all layers, but it's a non-trivial task on its own. Would like to hear thoughts from the team!

Reply:
So don't worry, It looks good to me to proceed in this way 👍

Reply:
Sorry, I'm late! 

I agree that we shouldn't import `Storage` here. `WooCommerce` should only reference `Yosemite`. We're still on a year-long crusade of removing `Storage` references from `WooCommerce` and we're not done yet! 😅 

From the [design doc](https://github.com/woocommerce/woocommerce-ios/blob/develop/docs/architecture-overview.md): 

> WooCommerce:  Our main target, which is expected to **only** interact with the entire stack thru the Yosemite.framework.

Jaclyn's idea of a separate Foundation framework sounds like a good option to try.

Reply:
Extracting those predicates into a separate package should be fairly easy! But The work would be to analyze what other utilities would be a good fit for that new package!

Reply:
Nice, I wasn't sure how much effort it takes to create a separate package/framework! I created a separate issue to extract the utilities in https://github.com/woocommerce/woocommerce-ios/issues/3547 - feel free to update the list of files to share!